src/OSPSuite.Core/Journal/RelatedItemFactory.cs

```diff
@@ -39,22 +51,67 @@ public class RelatedItemFactory : IRelatedItemFactory
       {
          _executionContext.Load(relatedObject);
 
-         return new RelatedItem
+         var name = relatedObject.Name;
+         var data = _relatedItemSerializer.Serialize(relatedObject);
+         var itemType = _relatedItemTypeRetriever.TypeFor(relatedObject);
+
+
+         return createRelatedItem(name, itemType, data, relatedItem =>
+         {
+            relatedItem.Origin = _applicationConfiguration.Product;
+            relatedItem.IconName = relatedObject.Icon;
+            relatedItem.FullPath = _projectRetriever.CurrentProject.FilePath;
+            relatedItem.Discriminator = _applicationDiscriminator.DiscriminatorFor(relatedObject);
+            relatedItem.Description = _relatedItemDescriptionCreator.DescriptionFor(relatedObject);
+         });
+      }
+
+      public RelatedItem CreateFromFile(string fileFullPath)
+      {
+         var name = FileHelper.FileNameFromFileFullPath(fileFullPath);
+         var data = File.ReadAllBytes(fileFullPath);
+         var itemType = Constants.RELATIVE_ITEM_FILE_ITEM_TYPE;
+
+         return createRelatedItem(name, itemType, data, x =>
+         {
+            x.Origin = Origins.Other;
+            x.IconName = iconFromFileExtension(new FileInfo(fileFullPath).Extension).IconName;
+            x.FullPath = fileFullPath;
+            x.Discriminator = itemType;
+         });
+      }
+
+      private RelatedItem createRelatedItem(string name, string itemType, byte[] data, Action<RelatedItem> configurationFunc)
+      {
+         var relatedItem = new RelatedItem
          {
             CreatedAt = SystemTime.UtcNow(),
-            Name = relatedObject.Name,
-            ItemType = _relatedItemTypeRetriever.TypeFor(relatedObject),
+            Name = name,
+            ItemType = itemType,
             Content = new Content
             {
-               Data = _relatedItemSerializer.Serialize(relatedObject)
+               Data = data
             },
-            Origin = _applicationConfiguration.Product,
             Version = _applicationConfiguration.FullVersion,
-            IconName = relatedObject.Icon,
-            FullPath = _projectRetriever.CurrentProject.FilePath,
-            Discriminator = _applicationDiscriminator.DiscriminatorFor(relatedObject),
-            Description = _relatedItemDescriptionCreator.DescriptionFor(relatedObject)
          };
+
+         configurationFunc(relatedItem);
+         return relatedItem;
+      }
+
+      private ApplicationIcon iconFromFileExtension(string extension)
```

Comment:
Trying to be funky here to have different icons depending on file extensions so that you can quickly see what is which

Reply:
Should we also have explicit icons for some fore file types?
* Matlab (*.m, *.mat, *.fig)
* R (*.r, *.rd)
* pkml

Reply:
Sure